dependencies.yaml

```diff
@@ -431,18 +424,16 @@ dependencies:
               - *tensordict
           - {matrix: null, packages: [*pytorch_pip, *tensordict]}
       - output_types: [conda]
+        # PyTorch will stop publishing conda packages after 2.5.
+        # Consider switching to conda-forge::pytorch-gpu.
+        # Note that the CUDA version may differ from the official PyTorch wheels.
         matrices:
-          - matrix: {cuda: "12.1"}
-            packages:
-              - pytorch-cuda=12.1
-          - matrix: {cuda: "12.4"}
+          - matrix: {cuda: "12.*"}
             packages:
-              - pytorch-cuda=12.4
-          - matrix: {cuda: "11.8"}
+            - pytorch-gpu>=2.3=*cuda120*
```

Comment:
This is already using conda-forge, I think? `pytorch-gpu` is a `conda-forge` package, not a `pytorch` channel package. Also, the latest builds are built with CUDA 12.6. CUDA 12.0 is no longer used to build.

Reply:
For compatibility reasons we may want to stick to older builds of pytorch-gpu (built with `cuda120`) for now. We will hopefully be able to relax this in the future.

Reply:
Yes, this PR switches to `conda-forge::pytorch-gpu` since `pytorch` channel will discontinue.

> Also, the latest conda-forge builds are built with CUDA 12.6. CUDA 12.0 is no longer used to build.

> For compatibility reasons we may want to stick to older builds of pytorch-gpu (built with cuda120) for now. We will hopefully be able to relax this in the future.

Oh, I had not noticed that the most recent build (_306) is only against 12.6. I agree with keeping 12.0 for better backward compatibility. However, the CUDA 11 build seems missing. Do we have details on their build matrix?

Reply:
CUDA 11 builds were dropped recently. You may need an older version for CUDA 11 compatibility. I also saw this while working on https://github.com/rapidsai/cudf/pull/17475. `mamba search -c conda-forge "pytorch=*=cuda118*"` indicates the latest version with CUDA 11 support is 2.5.1 build 303. The latest is 2.5.1 build 306.

Reply:
For completeness, the latest CUDA 12.0 build was also 2.5.1 build 303.

Reply:
Got it, thanks. It shouldn't be a dealbreaker unless another test component ends up requiring a newer version of torch on CUDA 11 down the line.